File: backend/src/main/java/net/zorphy/backend/project/qwirkle/dto/GameState.java
package net.zorphy.backend.project.qwirkle.dto;

import net.zorphy.backend.project.qwirkle.dto.tile.BoardTile;
import net.zorphy.backend.project.qwirkle.dto.tile.StackTile;
import net.zorphy.backend.project.qwirkle.dto.tile.Tile;

import java.util.List;

public record GameState(
        List<Tile> hand,
        List<StackTile> stack,
        List<BoardTile> board,
        List<PositionInfo> openPositions
) {
}

File: frontend/src/app/dto/sites/qwirkle/GameState.ts
import {StackTile} from "./StackTile";
import {BoardTile} from "./BoardTile";
import {Tile} from "./Tile";
import {PositionInfo} from "./PositionInfo";
import {GameConfig} from "./GameConfig";

export interface GameState {
    gameConfig: GameConfig,
    currentPlayerTurn: number,
    hand: Tile[],
    stack: StackTile[],
    board: BoardTile[],
    openPositions: PositionInfo[]
}